contracts/vesting/Vesting.sol

```diff
@@ -90,12 +90,20 @@ contract Vesting is OwnableUpgradeable {
         uint256 _totalLocked = initialVestingAmount - initialUnlockedAmount;
         unlockUnit = _totalLocked / duration;
         remainder = _totalLocked - (unlockUnit * duration);
-        token.transferFrom(
+        status = VestingStatus.PREPARED;
+
+        uint256 beforeBal = token.balanceOf(address(this));
+        bool ret = token.transferFrom(
             _distributor,
             address(this),
             initialVestingAmount * 10**18
         );
```

Comment:
If the token does not return any value, the tx fails with an unhandled exception.

> Error: Transaction reverted: function returned an unexpected amount of data

Would be better to use `SafeERC20`.